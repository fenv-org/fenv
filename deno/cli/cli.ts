import { Command, ValidationError } from '@cliffy/command';
import { CommandException, FenvContext, OperationSystem } from '@fenv/lib';
import { writeTextLine } from '../lib/src/io/io.ts';
import * as init from './src/commands/init.ts';
import meta from '../meta.json' with { type: 'json' };

export async function main(
  { args, context }: {
    args: string[];
    context: FenvContext;
  },
): Promise<number> {
  try {
    await new Command()
      .name('fenv')
      .version(meta.version)
      .description('Simple flutter sdk version management')
      .globalEnv(
        'FENV_ROOT=<path:string>',
        'The root directory of the fenv installation. e.g. $HOME/.fenv',
      )
      .command(
        'init',
        init.command.action((options, args) =>
          init.handler(context, options, args)
        ),
      )
      .error(reportError)
      .meta('deno', Deno.version.deno)
      .meta('v8', Deno.version.v8)
      .parse(args);
    return 0;
  } catch (error) {
    if (error instanceof CommandException) {
      return error.code;
    } else {
      return 1;
    }
  }

  function reportError(error: Error): void {
    if (error instanceof ValidationError) {
      return;
    }
    writeTextLine(context.stderr, `ERROR: ${error.message}`);
  }
}

function detectOS(osName: string): OperationSystem {
  switch (osName) {
    case 'windows':
      return OperationSystem.WINDOWS;
    case 'darwin':
      return OperationSystem.MACOS;
    default:
      return OperationSystem.LINUX;
  }
}

if (import.meta.main) {
  const context = new FenvContext(
    Deno.stdout.writable,
    Deno.stderr.writable,
    detectOS(Deno.build.os),
    Deno.build.os !== 'windows' ? Deno.env.get('SHELL')! : '',
  );
  const statusCode = await main({
    args: Deno.args,
    context,
  });
  Deno.exit(statusCode);
}
