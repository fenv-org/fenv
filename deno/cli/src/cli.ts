import { Command } from '@cliffy/command';

async function main(args: string[]): Promise<void> {
}

if (import.meta.main) main(Deno.args);
