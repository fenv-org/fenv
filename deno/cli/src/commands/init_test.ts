import { assertEquals } from '@std/assert';
import $ from '@david/dax';
import { main } from '../../cli.ts';

Deno.test('temp', async () => {
  await main(['init', '-d']);
});
