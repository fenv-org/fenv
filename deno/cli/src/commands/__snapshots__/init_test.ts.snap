export const snapshot = {};

snapshot[`init without path mode: zsh 1`] = `
stdout:
'# Load fenv automatically by appending the following to
# ~/.zprofile (for login shells)
# and ~/.zshrc (for interactive shells) :

export FENV_ROOT="\$HOME/.fenv"
command -v fenv >/dev/null || export PATH="\$FENV_ROOT/bin:\$PATH"
eval "\$(fenv init -)"

# Restart your shell for the changes to take effect:

exec \$SHELL -l

'
stderr:
""
`;

snapshot[`init without path mode: bash 1`] = `
stdout:
'# Load fenv automatically by appending the following to
# ~/.bash_profile if it exists, otherwise ~/.profile (for login shells)
# and ~/.bashrc (for interactive shells) :

export FENV_ROOT="\$HOME/.fenv"
command -v fenv >/dev/null || export PATH="\$FENV_ROOT/bin:\$PATH"
eval "\$(fenv init -)"

# Restart your shell for the changes to take effect:

exec \$SHELL -l

'
stderr:
""
`;

snapshot[`init without path mode: fish 1`] = `
stdout:
"# Add fenv executable to PATH by running
# the following interactively:

set -Ux FENV_ROOT \$HOME/.fenv
fish_add_path \$FENV_ROOT/bin

# Load fenv automatically by appending
# the following to ~/.config/fish/conf.d/fenv.fish:

fenv init - | source

# Restart your shell for the changes to take effect:

exec \$SHELL -l

"
stderr:
""
`;

snapshot[`detectShell: zsh 1`] = `
stdout:
"FENV_SHELL_DETECT=zsh
"
stderr:
"shell zsh
"
`;
